isthmus/src/test/java/io/substrait/isthmus/NestedExpressionsTest.java

```diff
@@ -0,0 +1,76 @@
+package io.substrait.isthmus;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.substrait.dsl.SubstraitBuilder;
+import io.substrait.expression.Expression;
+import io.substrait.extension.DefaultExtensionCatalog;
+import io.substrait.extension.SimpleExtension;
+import io.substrait.relation.Rel;
+import io.substrait.type.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.calcite.rel.RelNode;
+import org.junit.jupiter.api.Test;
+
+class NestedExpressionsTest extends PlanTestBase {
+
+  protected static final SimpleExtension.ExtensionCollection defaultExtensionCollection =
+      DefaultExtensionCatalog.DEFAULT_COLLECTION;
+  protected SubstraitBuilder b = new SubstraitBuilder(defaultExtensionCollection);
+  SubstraitToCalcite substraitToCalcite = new SubstraitToCalcite(extensions, typeFactory);
+
+  io.substrait.expression.Expression literalExpression =
+      Expression.BoolLiteral.builder().value(true).build();
+  Expression.ScalarFunctionInvocation nonLiteralExpression = b.add(b.i32(7), b.i32(42));
+  Expression.ScalarFunctionInvocation nonLiteralExpression2 = b.add(b.i32(3), b.i32(4));
+
+  final List<Type> tableType = List.of(R.I32, R.FP32, N.STRING, N.BOOLEAN);
+  final Rel commonTable = b.namedScan(List.of("example"), List.of("a", "b", "c", "d"), tableType);
+  final Rel emptyTable = b.emptyScan();
+
+  @Test
+  void NestedListWithJustLiteralsTest() {
```

Comment:
General Java style point. Tests are methods, so their names should be in `camelCase`. Only constructor functions that share the name of the class are capitalized

```suggestion
  void nestedListWithJustLiteralsTest() {
```